tests/gui/modules/progress-status.spec.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const m = require('mochainon');
+const settings = require('../../../lib/gui/models/settings');
+const progressStatus = require('../../../lib/gui/modules/progress-status');
+
+describe('Browser: progressStatus', function() {
+
+  describe('.fromFlashState()', function() {
+
+    beforeEach(function() {
+      this.state = {
+        type: 'write',
+        percentage: 0,
+        eta: 15,
+        speed: 100000000000000
+      };
+
+      settings.set('unmountOnSuccess', true);
+    });
+
+    it('should report 0% if percentage == 0 but speed != 0', function() {
+      m.chai.expect(progressStatus.fromFlashState(this.state)).to.equal('0% Flashing');
+    });
```

Comment:
The tests in here look much neater than those you removed from `main.spec.js` - good job :+1: